apps/catalog/catalog/src/app/dashboard/avails/manage/manage.component.ts

```diff
@@ -0,0 +1,180 @@
+import { MatDialog } from '@angular/material/dialog';
+import { centralOrgId } from '@env';
+import { MatSnackBar } from '@angular/material/snack-bar';
+import { NavigationService } from '@blockframes/ui/navigation.service';
+
+
+const contractQuery = (titleId: string): QueryConstraint[] => [
+  where('titleId', '==', titleId),
+  where('type', '==', 'mandate'),
+];
+
+const salesQuery = (termId: string): QueryConstraint[] => [
+  where('parentTermId', '==', termId),
+  where('type', '==', 'sale'),
+];
+
+function isTerm(term: Partial<Term>): term is Term {
+  return term.contractId ? true : false;
+}
+
+@Component({
+  selector: 'catalog-manage-avails',
+  templateUrl: './manage.component.html',
+  styleUrls: ['./manage.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class CatalogManageAvailsComponent implements OnInit {
+  public availsForm = new AvailsForm();
+  public form = new NegotiationForm({ terms: [] });
+  private currentOrg$ = this.orgService.currentOrg$;
+  indexId = 1;
+  termColumns = {
+    'duration.from': 'Terms Start Date',
+    'duration.to': 'Terms End Date',
+    territories: 'Territories',
+    medias: 'Medias',
+    exclusive: 'Exclusivity',
+    languages: 'Versions',
+  };
+  public title$ = this.route.params.pipe(
+    pluck('titleId'),
+    switchMap((titleId: string) => this.titleService.valueChanges(titleId)),
+    shareReplay({ bufferSize: 1, refCount: true })
+  );
+
+  private mandates$ = this.title$.pipe(
+    switchMap(title => this.contractService.valueChanges(contractQuery(title.id))),
+    shareReplay({ bufferSize: 1, refCount: true })
+  );
+
+  private terms$ = this.mandates$.pipe(
+    map(contracts => contracts.flatMap(({ termIds }) => termIds)),
+    switchMap(termIds => this.termService.valueChanges(termIds)),
+    shareReplay({ bufferSize: 1, refCount: true })
+  );
+
+  constructor(
+    private titleService: MovieService,
+    private dynTitleService: DynamicTitleService,
+    private route: ActivatedRoute,
+    private navService: NavigationService,
+    private snackBar: MatSnackBar,
+    private contractService: ContractService,
+    private termService: TermService,
+    private dialog: MatDialog,
+    private orgService: OrganizationService,
+  ) { }
+
+  async ngOnInit() {
+    const promises = [
+      firstValueFrom(this.title$),
+      firstValueFrom(this.terms$)
+    ] as const;
+    const [movie, terms] = await Promise.all(promises);
+    const pageTitle = `Manage avails of ${movie.title.international}`;
+    this.dynTitleService.setPageTitle(pageTitle);
+    this.form.hardReset({ terms });
+  }
+
+
+  openDetails(terms: string, scope: Scope) {
+    this.dialog.open(DetailedTermsComponent, { data: createModalData({ terms, scope }), autoFocus: false });
+  }
+
+  async areSalesAttachedToTerms(ids: string[]) {
+    const sales = await firstValueFrom(combineLatest(
+      ids.map(id => this.contractService.valueChanges(salesQuery(id)))
+    ));
```

Comment:
Something like this to make it more readable
```suggestion
    const queries = ids.map(id => this.contractService.valueChanges(salesQuery(id)));
    const sales = await firstValueFrom(combineLatest(queries));
```